arangod/RocksDBEngine/RocksDBCollectionMeta.cpp

```diff
@@ -111,27 +106,17 @@ Result RocksDBCollectionMeta::placeBlocker(uint64_t trxId, rocksdb::SequenceNumb
  *              earlier `placeBlocker` call)
  */
 void RocksDBCollectionMeta::removeBlocker(uint64_t trxId) {
-  WRITE_LOCKER(locker, _blockerLock);
-  auto it = _blockers.find(trxId);
-  if (ADB_LIKELY(_blockers.end() != it)) {
-    auto cross = _blockersBySeq.find(std::make_pair(it->second, it->first));
-    TRI_ASSERT(_blockersBySeq.end() != cross);
-    if (ADB_LIKELY(_blockersBySeq.end() != cross)) {
-      _blockersBySeq.erase(cross);
-    }
-    _blockers.erase(it);
-  }
+  _blockers.erase(trxId);
 }
 
 /// @brief updates and returns the largest safe seq to squash updated against
-rocksdb::SequenceNumber RocksDBCollectionMeta::committableSeq() const {
-  READ_LOCKER(locker, _blockerLock);
+rocksdb::SequenceNumber RocksDBCollectionMeta::committableSeq() {
   // if we have a blocker use the lowest counter
-  if (!_blockersBySeq.empty()) {
-    auto it = _blockersBySeq.begin();
-    return it->first;
+  auto res = std::numeric_limits<rocksdb::SequenceNumber>::max();
```

Comment:
I am not sure we want to perform a linear scan here

Reply:
I figured you would catch that. :) From what I figured `_blockers` did not have to be sorted, but `_blockersBySeq` did, so that `committableSeq()` could simply take the first entry. My impression from what I found in the code (correct me if I am wrong) is that calls to `committableSeq()` happen rather seldom. So I figured that the reduced cost for placing/removing blockers outweigh the increased cost of `committableSeq()`.
BTW: yes, this is a linear scan, but I would not expect the number of blockers to be that high (say, several thousands), and the design of the hash-map is highly cache aware, so the scan should be very efficient.